spec/features/merchant/discount_spec.rb

```diff
@@ -0,0 +1,189 @@
+require 'rails_helper'
+
```

Comment:
I have wondered if I should create a separate file for different discount functionality, but this follows the existing file structure. Future tests for applying discount to cart will be in a separate file.